From: Mike Turquette <mturquette@ti.com>
Date: Mon, 25 Apr 2011 19:44:22 +0000 (-0500)
Subject: OMAP4: Clock: use clk fwk apis in DPLL cascading
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=5934de1f0301914c64b1786a810cd2ded64ec38f

OMAP4: Clock: use clk fwk apis in DPLL cascading

To make use of the new clock notifiers when entering an exiting DPLL
cascading it is necessary to use the top-level clk_set_rate APIs.  This
patch replaces uses of omap3_noncore_dpll_set_rate with the more generic
clk_set_rate.  This fixes the issue of clock notifiers under DPLL_IVA &
DPLL_PER not getting propagated.

Change-Id: I2122353661b88d6bc011d8b6c597eb6202ccd5a4
Signed-off-by: Mike Turquette <mturquette@ti.com>
---

Index: kernel/arch/arm/mach-omap2/dpll-44xx.c
===================================================================
--- kernel.orig/arch/arm/mach-omap2/dpll-44xx.c	2011-05-21 00:42:36.330846960 +0900
+++ kernel/arch/arm/mach-omap2/dpll-44xx.c	2011-05-21 00:43:03.270847085 +0900
@@ -762,7 +762,7 @@
 
 	/* bypass DPLL_IVA */
 	state.dpll_iva_ck_rate = dpll_iva_ck->rate;
-	ret = omap3_noncore_dpll_set_rate(dpll_iva_ck,
+	ret = clk_set_rate(dpll_iva_ck,
 			dpll_iva_ck->dpll_data->clk_bypass->rate);
 	if (ret) {
 		pr_err("%s: DPLL_IVA failed to enter Low Power bypass\n",
@@ -773,7 +773,7 @@
 
 	/* bypass DPLL_PER */
 	state.dpll_per_ck_rate = dpll_per_ck->rate;
-	ret = omap3_noncore_dpll_set_rate(dpll_per_ck,
+	ret = clk_set_rate(dpll_per_ck,
 			dpll_per_ck->dpll_data->clk_bypass->rate);
 	if (ret) {
 		pr_debug("%s: DPLL_PER failed to enter Low Power bypass\n",
@@ -938,12 +938,12 @@
 	opp_disable(state.mpu_opp);
 
 	/* lock DPLL_IVA */
-	ret = omap3_noncore_dpll_set_rate(dpll_iva_ck, state.dpll_iva_ck_rate);
+	ret = clk_set_rate(dpll_iva_ck, state.dpll_iva_ck_rate);
 	if (ret)
 		pr_err("%s: DPLL_IVA failed to relock\n", __func__);
 
 	/* lock DPLL_PER */
-	ret = omap3_noncore_dpll_set_rate(dpll_per_ck, state.dpll_per_ck_rate);
+	ret = clk_set_rate(dpll_per_ck, state.dpll_per_ck_rate);
 	if (ret)
 		pr_err("%s: DPLL_PER failed to relock\n", __func__);
 
